payment-service/src/main/java/com/hedvig/paymentservice/services/adyen/util/AdyenMerchantPicker.kt

```diff
@@ -0,0 +1,95 @@
+import com.hedvig.paymentservice.serviceIntergration.underwriterClient.UnderwriterService
+import com.hedvig.paymentservice.services.adyen.dtos.AdyenMerchantInfo
+import com.neovisionaries.i18n.CountryCode
+import com.neovisionaries.i18n.CurrencyCode
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import com.hedvig.paymentservice.serviceIntergration.productPricing.dto.Market as ProductPricingMarket
+
+@Component
+class AdyenMerchantPicker(
+  val memberService: MemberService,
+  val underwriterService: UnderwriterService,
+  val productPricingService: ProductPricingService,
+  val memberRepository: MemberRepository,
+  val merchantAccounts: MerchantAccounts
+) {
+  fun getAdyenMerchantInfo(memberId: String): AdyenMerchantInfo {
+    val marketInfo = getMerchantFromMember(memberId)
+      ?: getMarketFromContract(memberId)
+      ?: getMarketFromQuote(memberId)
+      ?: getMarketFromPickedLocale(memberId)
+      ?: throw NullPointerException("Could not determine market for member: $memberId")
+
+    return AdyenMerchantInfo(
+      account = merchantAccounts.merchantAccounts!![marketInfo.name] ?: error("Cannot fetch merchant account"),
+      countryCode = marketInfo.countryCode,
+      currencyCode = marketInfo.currencyCode
+    )
+  }
+
+  private fun getMerchantFromMember(memberId: String): Market? {
+    val memberMaybe = memberRepository.findById(memberId)
+
+    if (!memberMaybe.isPresent) return null
+
+    val member = memberMaybe.get()
+
+    val adyenMerchantAccount: String = member.adyenMerchantAccount ?: return null
+
+    return Market.valueOf(
+      merchantAccounts.merchantAccounts!!
+        .filterValues { it == adyenMerchantAccount }.keys.first()
+    )
+  }
+
+  private fun getMarketFromContract(memberId: String): Market? {
+    return try {
+      val contractMarketInfo = productPricingService.getContractMarketInfo(memberId)
+      when (contractMarketInfo.market) {
+        ProductPricingMarket.SWEDEN -> Market.SWEDEN
+        ProductPricingMarket.NORWAY -> Market.NORWAY
+        ProductPricingMarket.DENMARK -> Market.DENMARK
+      }
+    } catch (exception: Exception) {
+      logger.error("Cannot find market from contact for member $memberId")
+      null
+    }
+  }
+
+  private fun getMarketFromQuote(memberId: String): Market? {
+    return try {
+      val quoteMarketInfo = underwriterService.getMarketFromQuote(memberId)
+      Market.valueOf(quoteMarketInfo.market)
+    } catch (exception: Exception) {
+      logger.error("Cannot find market from quotes for member $memberId")
+      null
+    }
+  }
+
+  private fun getMarketFromPickedLocale(receiverMemberId: String): Market? {
+    return when (memberService.getPickedLocale(receiverMemberId)) {
+      "sv_SE" -> Market.SWEDEN
+      "en_SE" -> Market.SWEDEN
+      "nb_NO" -> Market.NORWAY
+      "en_NO" -> Market.NORWAY
+      else -> null
+    }
```

Comment:
I know `da_DK` and `en_DK` in member-service yet but I think we can map them here either way. It is easy to forget otherwise
```suggestion
    return when (memberService.getPickedLocale(receiverMemberId)) {
      "sv_SE", "en_SE" -> Market.SWEDEN
      "nb_NO", "en_NO" -> Market.NORWAY
      "da_DK", "en_DK" -> Market.DENMARK
      else -> null
    }
```